include/fastrtps/rtps/messages/RTPSMessageGroup.h

```diff
@@ -79,31 +79,103 @@ class RTPSMessageGroup
+         */
         bool add_data(const CacheChange_t& change, const std::vector<GUID_t>& remote_readers,
                 const LocatorList_t& locators, bool expectsInlineQos);
 
+        /**
+         * Adds a DATA_FRAG message to the group.
+         * @param change Reference to the cache change to send.
+         * @param fragment_number Index (1 based) of the fragment to send.
+         * @param remote_readers List of destination GUIDs.
+         * @param locators List of destination locators.
+         * @param expectsInlineQos True when one destination is expecting inline QOS.
+         * @return True when message was added to the group.
+         */
         bool add_data_frag(const CacheChange_t& change, const uint32_t fragment_number,
                 const std::vector<GUID_t>& remote_readers, const LocatorList_t& locators,
                 bool expectsInlineQos);
 
+        /**
+         * Adds a HEARTBEAT message to the group.
+         * @param remote_readers List of destination GUIDs.
+         * @param firstSN First available sequence number.
+         * @param lastSN Last available sequence number.
+         * @param count Counting identifier.
+         * @param isFinal Should final flag be set?
+         * @param livelinessFlag Should liveliness flag be set?
+         * @param locators List of destination locators.
+         * @return True when message was added to the group.
+         */
         bool add_heartbeat(const std::vector<GUID_t>& remote_readers, const SequenceNumber_t& firstSN,
                 const SequenceNumber_t& lastSN, Count_t count,
                 bool isFinal, bool livelinessFlag, const LocatorList_t& locators);
 
+        /**
+         * Adds a GAP message to the group.
+         * @param changesSeqNum Set of missed sequence numbers.
+         * @param remote_readers List of destination GUIDs.
+         * @param locators List of destination locators.
+         * @return True when message was added to the group.
+         */
         bool add_gap(std::set<SequenceNumber_t>& changesSeqNum, const std::vector<GUID_t>& remote_readers,
                 const LocatorList_t& locators);
 
+        /**
+         * Adds a ACKNACK message to the group.
+         * @param remote_writers List of destination GUIDs (note: only first one will be used).
+         * @param SNSet Set of received sequence numbers.
+         * @param count Counting identifier.
+         * @param finalFlag Should final flag be set?
+         * @param locators List of destination locators.
+         * @return True when message was added to the group.
+         */
         bool add_acknack(const std::vector<GUID_t>& remote_writers, SequenceNumberSet_t& SNSet,
                 int32_t count, bool finalFlag, const LocatorList_t& locators);
 
+        /**
+         * Adds a NACKFRAG message to the group.
+         * @param remote_writers List of destination GUIDs (note: only first one will be used).
```

Comment:
This function's documentation is not according to the real function.

Reply:
Fixed!